File: Data Structures & Algorithms/longest-common-prefix/submission-0.java
class Solution {
    public String longestCommonPrefix(String[] strs) {
        // slow way
        String ans = "";
        String check = strs[0];
        for (String curr : strs) {
            if (check.length() > curr.length()) check = curr;
        }
        boolean state = true;
        String longest = "";
        for (int i = 0; i < check.length(); i++){
            longest = check.substring(0, i + 1);
            for (int j = 0; j < strs.length; j++){
                System.out.println(longest.equals(strs[j].substring(0, i + 1)));
                if (!longest.equals(strs[j].substring(0, i + 1))) {
                    state = false;
                    break;
                }
            }
            if (!state) {
                break;
            }
            ans = longest;
        }
        return ans;
    }
}

File: Data Structures & Algorithms/max-water-container/submission-0.py
class Solution:
    def maxArea(self, heights: List[int]) -> int:
        left = 0
        right = len(heights) - 1
        ans = 0
        while left < right:
            curr = 0
            nI = ""
            if heights[left] >= heights[right]:
                curr = heights[right]
                nI = "r"
            else:
                curr = heights[left]
                nI = "l"
            if curr *  (right - left) > ans:
                ans = curr * (right - left)
            if nI == "r":
                right -=1
            else: 
                left += 1
        return ans

            